lib/elastic_apm/transaction.rb

```diff
@@ -26,28 +30,20 @@ def initialize(
       @context = context || Context.new # TODO: Lazy generate this?
       Util.reverse_merge!(@context.tags, tags) if tags
 
-      if trace_context
-        @parent_id = trace_context.span_id
-        @trace_context = trace_context
-      else
-        @trace_context = TraceContext.for_transaction(sampled: sampled)
-      end
+      @trace_context = trace_context ||
+                       TraceContext.for_transaction(sampled: sampled)
```

Comment:
we almost don't need the `for_transaction()` now? Could be just `new(sampled: sampled)` and the result would be the same?